### DISCUSSION TITLE
[ZK Stack] future transaction tries to replace pending

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#713] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I was trying to deploy Zk stack on sepolia and I encountered this error:
Im using Alchemy's growth plan.

Edit: Tutorial Im following:  https://code.zksync.io/tutorials/custom-zk-chain

```
Estimated amount required: 18.026436340496801033 ETH      
                                                         
==========================                               
                                                         
Error:                                                   
Failed to send transaction                               
                                                         
Context:                                                                                                           
- server returned an error response: error code -32000: future transaction tries to replace pending                 
│                            
■  Command failed to run                                 
│                                                        
■    Status:                                                                                                       
│      exit status: 1                                                                                              
│    Stdout:                                             
│                                                        
│                                                        
│    Stderr:                                             
│      
│  
■  Command failed to run: forge script deploy-scripts/DeployL1.s.sol --legacy --ffi --rpc-url=https://eth-sepolia.g.alchemy.com/v2/W-FEBNPHBcJk5aHyzzzzzzzzzzz
GQIOf --private-key=623414291b501ea3f545b6be4306e1ef4b17d097f0fce6d995e5561xxxxxxxx --broadcast
│  
│  
▲    0: Command failed to run: forge script deploy-scripts/DeployL1.s.sol --legacy --ffi --rpc-url=https://eth-sepolia.g.alchemy.com/v2/W-FEBNPHBcJk5aHyzzzzzzzzzzz
ExpeyGQIOf --private-key=623414291b501ea3f545b6be4306e1ef4b17d097f0fce6d995e5561xxxxxxxx --broadcast

```

### Expected Behavior

Contracts get deployed

### Current Behavior

Contracts not getting deployed.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
The error code -32000, indicating "initcode too large," typically occurs when the size of the contract's initialization code exceeds the block gas limit or the specific limits set by the Ethereum client or testnet you are using. In your case, deploying on the Holesky testnet using Alchemy, you might be hitting such a limit.

Here are a few steps you can take to resolve this issue:

1. **Optimize Contract Code**: Ensure your contract code is optimized to reduce its size. This can involve minimizing the use of large libraries, or splitting your contract into smaller, more manageable pieces.

2. **Change RPC Provider**: Sometimes, the issue might be related to the specific RPC provider's settings or limitations. Trying a different RPC provider could potentially resolve the issue.

3. **Adjust Compiler Settings**: Use compiler optimization settings to reduce the bytecode size. For instance, adjusting the settings in your `solc` compiler for higher optimization levels might help.

4. **Refactor the Contract**: If the contract is inherently large, consider refactoring it into smaller contracts to ensure each deployed piece is within size limits.

For more detailed guidance on handling large contracts and optimization, you might find the following resources helpful:
- [Contract size limit and format of bytecode hash](https://docs.zksync.io/build/developer-reference/ethereum-differences/contract-deployment#contract-size-limit-and-format-of-bytecode-hash)
- [Compilation and optimization tips](https://docs.zksync.io/build/tooling/foundry/migration-guide/compilation)

If these steps do not resolve the issue, it may be beneficial to reach out to the support team of your RPC provider (Alchemy in this case) for more tailored assistance.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by iscxyz** [comment]
Closing down as duplicate. See [this](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/714) one for updates

**Reply by sachanayush47** [reply]
> Closing down as duplicate. See [this](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/714) one for updates

Both issues are not same. Both involves different network, different error message and different contract.

**Comment by sachanayush47** [comment, marked as answer]
On our investigation we found that we trying to deploy chain and ecosystem again and again with the same wallets which is causing some nonce calculation issue. To fix this issue, either use new wallets or make a dummy transactions from all wallets or wait around 6 hours to try again from same wallets.

**Reply by bxpana** [reply]
Thanks for updating the community!
